packages/dds/tree/src/core/tree/pathTree.ts

```diff
@@ -5,6 +5,12 @@
 
 import { FieldKey } from "./types";
 
+/**
+ * Identical to {@link UpPath2}, but a duplicate declaration is needed to make the default type parameter compile.
+ * @alpha
+ */
+export type UpPath2 = UpPath;
```

Comment:
I don't know of a clean way around this. You can consider something a little clunky [like this](https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgKoAcAKcwAsCSAPACrZQTgB8yA3gL4BQEAHugPZRjJgCe6KGbHhJkKXALzIAriAAmEGKAizqkwTgIi45cMhaQ5AZ2lyFS2cgD8aLBuQAuZKW1jKAbiA), since `undefined` is already not a valid kind of parent.

But it might be better to simply [not have a default generic parameter, and rename `UpPath2` to `DefaultUpPath`](https://www.typescriptlang.org/play?#code/KYDwDg9gTgLgBDAnmYcAiwBmBDArgGxgFUwAFbGACzgF44TyqAeDHA4siygPgG4AoUJFhwAlgDsYwKDgDGqBlyYAVclGCTucAN4BfIA) or something.

Reply:
I have renamed it to make it look less like a hack.